lib/rspec/rails/matchers/have_http_status.rb

```diff
@@ -283,6 +287,12 @@ def failure_message_when_negated
             "expected the response not to have #{type_message} but it was #{actual}"
           end
 
+        protected
+
+          def check_expected_status(test_response, expected)
+            test_response.send("#{expected}?")
+          end
```

Comment:
Basically this would become something like:

```Ruby
if if 5 < ::Rails::VERSION::MAJOR || (::Rails::VERSION::MAJOR == 5 && 2 <= ::Rails::VERSION::MINOR)
  def check_expected_status(test_response, expected)
    test_response.send("#{expected}?")
  end
else
  RESPONSE_METHODS = {
    success: 'successful',
    error: 'server_error',
    missing: 'not_found'
  }.freeze

  def check_expected_status(test_response, expected)
     test_response.send("#{RESPONSE_METHODS.fetch(response_symbol, response_symbol)}?")
  end
end
```